linkerd/protocol/thrift/src/main/scala/com/twitter/finagle/buoyant/linkerd/ThriftServerPrep.scala

```diff
@@ -21,13 +21,15 @@ object ThriftServerPrep {
     ): ServiceFactory[Array[Byte], Array[Byte]] = {
       val Label(label) = params[Label]
       val Thrift.param.ProtocolFactory(pf) = params[Thrift.param.ProtocolFactory]
-      val payloadSize = new PayloadSizeFilter[Array[Byte], Array[Byte]](
-        params[param.Stats].statsReceiver, _.length, _.length
-      )
       val uncaughtExceptionsFilter = new UncaughtAppExceptionFilter(pf)
-      val ttwitter = new TTwitterServerFilter(label, pf)
 
-      payloadSize.andThen(ttwitter).andThen(uncaughtExceptionsFilter).andThen(next)
+      next.map { service =>
+        val payloadSize = new PayloadSizeFilter[Array[Byte], Array[Byte]](
+          params[param.Stats].statsReceiver, _.length, _.length
+        )
+        val ttwitter = new TTwitterServerFilter(label, pf)
+        payloadSize.andThen(ttwitter).andThen(uncaughtExceptionsFilter).andThen(service)
```

Comment:
`uncaughtExceptionsFilter` is outside of the map.  `payloadSize` being inside is cargo-cult from finagle https://github.com/twitter/finagle/blob/695694ea710bda551ed3da7702a2bd5776ce4e20/finagle-thrift/src/main/scala/com/twitter/finagle/thrift/ThriftServerPreparer.scala

I'm not sure if there's a good reason but I feel like mirroring finagle is safer.